src/main/java/org/prnhs/javaee/swim/services/PlanPracticeService.java

```diff
@@ -15,67 +17,73 @@
 @Service
 public class PlanPracticeService {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(PlanPracticeService.class);
+
     @Autowired
     private PlanPracticeDao dao;
 
     public PlanPracticeDto save(PlanPracticeDto planPracticeDto) {
 
         if (planPracticeDto == null) {
+            LOGGER.error("'Plan Practice' is missing!");
             throw new IllegalArgumentException("You have to provide the Plan Practice!");
         }
 
         PlanPractice planPractice = dao.findOne(planPracticeDto.getId());
 
         if (planPractice == null) {
+            LOGGER.debug("'Given 'Plan Practice' is not found, creating a new one.");
             planPractice = PlanPracticeTranslator.toEntity(planPracticeDto);
         } else {
+            LOGGER.debug("Updating {}", planPracticeDto);
             planPractice.setLength(planPracticeDto.getLength());
             planPractice.setMultiple(planPracticeDto.getMultiple());
             planPractice.setExercise(planPracticeDto.getExercise());
             planPractice.setSplit(planPracticeDto.getSplit());
         }
 
         planPractice = dao.save(planPractice);
+        LOGGER.debug("{} saved successfully", planPracticeDto);
```

Comment:
might be better to say 'saved succesfful with result {}'  because you never know how big that object is and your message will get lost at the end.